src/cosalib/gcp.py

```diff
@@ -62,7 +62,18 @@ def gcp_run_ore(build, args):
         '--write-url', urltmp,
     ])
 
+    if args.fcos:
+        ore_args.extend(['--fcos'])
+    if args.family:
+        ore_args.extend(['--family', args.family])
+
     run_verbose(ore_args)
+
+    # Here is where I would do something like:
+    if args.deprecate:
+        run_verbose(ore gcloud deprecate-image --image-name gcp_name)
```

Comment:
so we could either encode a seperate call to ore here if the user requested the image to be created in the deprecated state OR we could modify `ore gcloud upload` to accept a `--deprecated` argument as well.

Reply:
So to my understanding there are two different times when we'd want to deprecate an image:

1) When we first upload it
2) To mark the old active image as deprecated in preparation for a new active image

For the first case that should probably be the default behavior performed by `ore gcloud upload` (with a flag to turn off that behavior). The second should always be done as a separate operation so that it's more obvious and intentional that an image (which is not the one being uploaded) is being marked as deprecated.

Reply:
> So to my understanding there are two different times when we'd want to deprecate an image:
> 
>     1. When we first upload it
> 
>     2. To mark the old active image as deprecated in preparation for a new active image

correct. with a slight addition to `2`. For `1` it happens when we
first do the upload (during the pipeline run). For `2` it would happen
during the release and we'd need to do two things:

- set deprecation status to "DEPRECATED" for the image that is currently latest in image family
- set deprecation status to "ACTIVE" of new image

So we'd make two calls to do that.

> 
> 
> For the first case that should probably be the default behavior performed by `ore gcloud upload` (with a flag to turn off that behavior). The second should always be done as a separate operation so that it's more obvious and intentional that an image (which is not the one being uploaded) is being marked as deprecated.

Since uploading an image that you immediately want to deprecate is kind of a special weird use case we're implementing I thought it would be more appropriate to leave it as not the default and explicitly pass `--deprecated` or something in the pipeline. For example, I think RHCOS uses this functionality and I don't think we want to change the defaults there right now.

Reply:
> correct. with a slight addition to `2`. For `1` it happens when we
> first do the upload (during the pipeline run). For `2` it would happen
> during the release and we'd need to do two things:
> 
> * set deprecation status to "DEPRECATED" for the image that is currently latest in image family
> * set deprecation status to "ACTIVE" of new image
> 
> So we'd make two calls to do that.

I'm personally alright with it being two calls (with the caveat that it might make sense to have it be one COSA command which performs both ore actions).

> Since uploading an image that you immediately want to deprecate is kind of a special weird use case we're implementing I thought it would be more appropriate to leave it as not the default and explicitly pass `--deprecated` or something in the pipeline. For example, I think RHCOS uses this functionality and I don't think we want to change the defaults there right now.

That's fair. We could potentially do something like switching the default for FCOS (since we're already passing a flag directly indicating if it's an FCOS upload) but I'd rather stick to just directly passing it in the pipeline.

Reply:
That's exactly what I was thinking. Cool I'll start down that path.